src/storage/mod.rs

```diff
@@ -120,7 +120,7 @@ pub type Callback<T> = Box<dyn FnOnce(Result<T>) + Send>;
 /// encoding and appending timestamp.
 pub struct Storage<E: Engine, L: LockManager> {
     // TODO: Too many Arcs, would be slow when clone.
-    engine: E,
+    pub engine: E,
```

Comment:
🤔 

`storage.engine` is only used in tests. can you instead create a function like


```rust
#[cfg(test)]
pub fn xxxxx(&self) -> Result<E::Snap> {
    self.engine.snapshot(Default::default())
}
```

or investigate whether using `storage.get_engine().snapshot(Default::default())` in the tests is ok.

Reply:
OK